agent/src/main/scala/za/co/absa/atum/agent/model/MeasurementBuilder.scala

```diff
@@ -16,19 +16,40 @@
 
 package za.co.absa.atum.agent.model
 
+import za.co.absa.atum.agent.exception.AtumAgentException.MeasurementException
 import za.co.absa.atum.model.dto.{MeasureDTO, MeasureResultDTO, MeasurementDTO}
-import za.co.absa.atum.model.dto.MeasureResultDTO.TypedValue
 
 /**
  * This object provides a functionality to convert a measurement to its DTO representation.
  */
 private [agent] object MeasurementBuilder {
 
-  private [agent] def buildMeasurementDTO(measurement: Measurement): MeasurementDTO = {
-    val measureName = measurement.measure.measureName
-    val measureDTO = MeasureDTO(measureName, measurement.measure.controlColumns)
-    val measureResultDTO = MeasureResultDTO(TypedValue(measurement.resultValue.toString, measurement.resultType))
+  private[agent] def validateMeasureUniqueness(measurements: Set[Measurement]): Unit = {
```

Comment:
Do you need to really validate this? Isn't that what `Set` itself does as it uses equals and hashCode methods.